packages/cli/src/databases/migrations/common/1717498465932-RefactorExecutionIndices.ts

```diff
@@ -0,0 +1,92 @@
+import type { MigrationContext, ReversibleMigration } from '@/databases/types';
+
+/**
+ * Add new indices:
+ *
+ * - `status, startedAt` for `ExecutionRepository.findManyByRangeQuery` (default query)
+ * - `workflowId, status, startedAt` for `ExecutionRepository.findManyByRangeQuery` (filter query)
+ * - `waitTill, status` for `ExecutionRepository.getWaitingExecutions`
+ * - `stoppedAt, deletedAt, status` for `ExecutionRepository.softDeletePrunableExecutions`
+ *
+ * Remove unused indices in sqlite:
+ *
+ * - `stoppedAt` (duplicate with different casing)
+ * - `waitTill`
+ * - `status, workflowId`
+ *
+ * Remove unused indices in MySQL:
+ *
+ * - `status`
+ *
+ * Remove unused indices in all DBs:
+ *
+ * - `waitTill, id`
+ * - `workflowId, id`
+ *
+ * Remove incomplete index in all DBs:
+ *
+ * - `stopped_at` (replaced with composite index)
+ *
+ * Keep index as is:
+ *
+ * - `deletedAt` for query at `ExecutionRepository.hardDeleteSoftDeletedExecutions`
+ */
+export class RefactorExecutionIndices1717498465932 implements ReversibleMigration {
+	async up({ schemaBuilder, isPostgres, isSqlite, isMysql }: MigrationContext) {
+		await schemaBuilder.createIndex('execution_entity', ['status', 'startedAt']);
+		await schemaBuilder.createIndex('execution_entity', ['workflowId', 'status', 'startedAt']);
+		await schemaBuilder.createIndex('execution_entity', ['waitTill', 'status']);
+		await schemaBuilder.createIndex('execution_entity', ['stoppedAt', 'deletedAt', 'status']);
```

Comment:
These are the names, I find them clear enough, what would you suggest?

![Capture 2024-07-30 at 17 11 44@2x](https://github.com/user-attachments/assets/3b67ce52-648d-4c5c-9531-94491f4b7460)

Reply:
I meant name them explicitly, so it's obvious from the code what the index names will be, instead of letting the ORM generate the names. But those are also fine as long as they are stable across all different DBs